crypto/bio/bss_core.c

```diff
@@ -101,6 +101,9 @@ static int bio_core_free(BIO *bio)
 {
     BIO_CORE_GLOBALS *bcgbl = get_globals(bio->libctx);
 
+    if (bcgbl == NULL)
+        return -1;
```

Comment:
Probably this should be return 0 on failure. All the other destroy functions that I could see in "crypto/bio" return 0 on failure. It doesn't really "matter" because when the BIO_free() functions calls this the return value is ignored anyway!! But for consistency we should probably do what the other implementations do. Also I note that BIO_free() is documented as returning 1 for success and 0 for failure, so it makes sense for the destroy function to do the same.

Reply:
```suggestion
        return 0;
```